templates/file-content-redaction.yaml.j2

```diff
@@ -0,0 +1,43 @@
+# file-conrent-redaction.yaml
+# Docs: https://access.redhat.com/articles/4511681
+---
+# Omit lines from files and command output using parameters listed here.
+# Lines matching the parameters specified will be omitted
+#   in the order that the parameters are given, e.g.,
+#
+# patterns:
+#   - "example_string_1"
+#   - "example_string_2"
+#
+# Lines containing "example_string_1" or "example_string_2" will be
+# omitted from output.
+#
+# To use regular expressions, wrap the array with "regex" like the following example:
+#
+# patterns:
+#   regex:
+#     - abc.*
+#     - localhost[[:digit:]]
+#
+# Lines matching these regular expressions will be omitted
+# from output.
+# NOTE: You cannot mix plain string matching and regular expression matching.
+{% if rhc_insights.file_content_redaction.keywords %}
+keywords:
+{% for keyword in rhc_insights.file_content_redaction.keywords %}
+  - {{ keyword }}
```

Comment:
For patterns (with regex) that would be possible as the examples in [YAML-style denylist configuration for Red Hat Insights Client](https://access.redhat.com/articles/4511681) show characters that need to be escaped.

For keywords, I tend to escape them just to be sure. Do you agree?

Reply:
This is tricky - afaik there is no "regex quote" or "regex escape" filter in Jinja or Ansible (like there is for shell metacharacters and the `quote` filter).  If you use `- "{{ pattern }}"` I think this will cause problems if `pattern` contains `"` - same with `'{{ pattern }}'` and `'` - however, this might be safe:
```yaml
  - >-
   {{ pattern }}
```